Microsoft.Toolkit.Uwp.UI.Controls/Graph/PeoplePicker/PeoplePicker.Events.cs

```diff
@@ -0,0 +1,122 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Graph;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls.Graph
+{
+    /// <summary>
+    /// Defines the events for the <see cref="PeoplePicker"/> control.
+    /// </summary>
+    public partial class PeoplePicker : Control
+    {
+        private void ClearAndHideSearchResultListBox()
+        {
+            SearchResultList.Clear();
+            _searchResultListBox.Visibility = Visibility.Collapsed;
+        }
+
+        private async void SearchBox_OnTextChanged(object sender, TextChangedEventArgs e)
+        {
+            var textboxSender = (TextBox)sender;
+            string searchText = textboxSender.Text.Trim();
+            if (string.IsNullOrWhiteSpace(searchText))
+            {
+                ClearAndHideSearchResultListBox();
+                return;
+            }
+
+            _loading.IsActive = true;
+
+            searchText = Regex.Replace(searchText, "[^0-9a-zA-Z .@]", string.Empty);
+            int cursorPosition = textboxSender.SelectionStart;
+            textboxSender.Text = searchText;
+            textboxSender.SelectionStart = cursorPosition;
+
+            try
+            {
+                var options = new List<QueryOption>
+                {
+                    new QueryOption("$search", searchText)
+                };
+                IUserPeopleCollectionPage peopleList = await GraphClient.Me.People.Request(options).GetAsync();
+
+                if (peopleList.Any())
+                {
+                    List<Person> searchResult = peopleList.Where(
+                        u => !string.IsNullOrWhiteSpace(u.UserPrincipalName)).ToList();
+
+                    // Remove all selected items
+                    foreach (Person selectedItem in Selections)
+                    {
+                        searchResult.RemoveAll(u => u.UserPrincipalName == selectedItem.UserPrincipalName);
+                    }
+
+                    SearchResultList = new ObservableCollection<Person>(SearchResultLimit > 0
```

Comment:
This is recreating an instance of the observable collection every keystroke. It should use the CLear and Add methods of the ObservableCollection, making the instance of the collection itself unique. The dependency property should initialize this with an empty ObservableCollection instance.

Reply:
@azchohfi Thanks! We have fixed this issue and make the collection unique.